src/main/java/org/jenkins/plugins/lockableresources/LockableResource.java

```diff
@@ -65,12 +65,6 @@
 	private String buildExternalizableId = null;
 	private long queuingStarted = 0;
 
-	/**
-	 * Only used when this lockable resource is tried to be locked by {@link LockStep},
-	 * otherwise (freestyle builds) regular Jenkins queue is used.
-	 */
-	private List<StepContext> queuedContexts = new ArrayList<StepContext>();
```

Comment:
However, once you have several resources, where would you store the queuedContexts? There is no longer one single resource (as in your implementation) to handle this.